pkg/controller/controller_test.go

```diff
@@ -152,6 +157,84 @@ func TestController(t *testing.T) {
 				},
 			},
 		},
+		"actual object is passed to the plugin": &testCase{
```

Comment:
I'm planning to add more unit tests for plugins to exercise a few other `if`s. But first this test needs some refactoring to de-duplicate common bits.